course_dashboard/answers_distribution_reports_manager/tasks.py

```diff
@@ -0,0 +1,143 @@
+import csv
+import datetime
+import json
+import xml.etree.ElementTree as ET
+
+from django.contrib.auth.models import User, Group
+from django.conf import settings
+
+from courseware.models import StudentModule
+from student.models import UserProfile
+from opaque_keys.edx.keys import UsageKey
+from xmodule.modulestore.django import modulestore
+
+from course_dashboard.answers_distribution import get_problem_module, add_ancestors_names_to_problem_module
+from fun import shared
+
+def cleanup_newlines(s):
+    """Makes sure all the newlines in s are representend by \r only."""
+    return s.replace("\r\n","\r").replace("\n","\r")
+
+def write_csv(header_row, data_rows, filename, course_id):
+    """write a CSV file from the contents of a datatable."""
+    shared.ensure_directory_exists(settings.ANSWERS_DISTRIBUTION_REPORTS_DIRECTORY, course_id.org, course_id.course)
+    
+    path = shared.get_path(settings.ANSWERS_DISTRIBUTION_REPORTS_DIRECTORY,
+                           course_id.org,
+                           course_id.course,
+                           filename)
+    
+    path = path.encode('utf-8')
+    
+    with open(path, "wb") as ofile:
+        writer = csv.writer(ofile, dialect='excel', quotechar='"', quoting=csv.QUOTE_ALL)
+        writer.writerow(header_row)
+        for datarow in data_rows:
+            encoded_row = [cleanup_newlines(unicode(s).encode('utf-8')) for s in datarow]
+            writer.writerow(encoded_row)
+            
+def create_header_row(quizz_size):
+     header_row = ['id', 'gender', 'year_of_birth', 'level_of_education']
+     
+     i = 0;
+     while i < quizz_size:
+         header_row.append( 'q' + str(i + 1))
+         i += 1
+     return header_row
+
+problem_types = ["multiplechoiceresponse", "choiceresponse" ,"stringresponse",
+                 "optionresponse", "numericalresponse", "customresponse"]
+
+def create_list_of_question_ids(organisation, course_number, problem_module_id, problem_module_size):
+     question_ids = []
+
+     i = 0;
+
+     while i < problem_module_size:
+          question_ids.append( "i4x-{}-{}-problem-{}_{}_1".format(organisation,course_number, problem_module_id, str((i + 2))))
+          i += 1
+     return question_ids
+
+
+
+def get_problem_module_size(problem_module):
+    """
+    Return the number of problem in the problem_module
+    """
+    
+    problem_module_as_xml = ET.fromstring((problem_module.data).encode('utf-8'))
+    problem_module_size = 0
```

Comment:
problem_module_size = len([element for element in problem_module_as_xml.iter() if element.tag in PROBLEM_TYPES])